qa/src/test/java/io/camunda/migrator/qa/history/HistoryMigrationSkippingTest.java

```diff
@@ -0,0 +1,383 @@
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+
+@TestPropertySource(properties = {
+    "logging.level.io.camunda.migrator.HistoryMigrator=DEBUG"
+})
+@ActiveProfiles("test")
+@ExtendWith({OutputCaptureExtension.class})
+public class HistoryMigrationSkippingTest extends HistoryMigrationAbstractTest {
+
+    @RegisterExtension
+    protected LogCapturer logs = LogCapturer.create().captureForLogger("io.camunda.migrator.HistoryMigrator");
+
+    @Autowired
+    protected DbClient dbClient;
+
+    @Autowired
+    private ManagementService managementService;
+
+    @BeforeEach
+    void ensureLoggingLevel() {
+        // Set the log level before each test to ensure it's applied after Spring context is ready
+        ch.qos.logback.classic.Logger logger = (ch.qos.logback.classic.Logger) org.slf4j.LoggerFactory.getLogger(HistoryMigrator.class);
+        logger.setLevel(Level.DEBUG);
+    }
+
+    @Test
+    public void shouldSkipElementsWhenProcessDefinitionIsSkipped() {
+        // given state in c7
+        deployer.deployCamunda7Process("userTaskProcess.bpmn");
+
+        for(int i = 0; i < 5; i++) {
+            runtimeService.startProcessInstanceByKey("userTaskProcessId");
+        }
+        for (Task task : taskService.createTaskQuery().list()) {
+            taskService.complete(task.getId());
+        }
+
+        // and the process definitions is manually set as skipped
+        String legacyId = repositoryService.createProcessDefinitionQuery().singleResult().getId();
+        dbClient.insert(legacyId, null, IdKeyMapper.TYPE.HISTORY_PROCESS_DEFINITION);
+
+        // when history is migrated
+        historyMigrator.migrate();
+
+        // then nothing was migrated
+        assertThat(searchHistoricProcessInstances("userTaskProcessId").size()).isEqualTo(0);
+
+        // and all elements for the definition were skipped
+        assertThat(dbClient.countSkippedByType(IdKeyMapper.TYPE.HISTORY_PROCESS_INSTANCE)).isEqualTo(5);
+        assertThat(dbClient.countSkippedByType(IdKeyMapper.TYPE.HISTORY_FLOW_NODE)).isEqualTo(15);
+        assertThat(dbClient.countSkippedByType(IdKeyMapper.TYPE.HISTORY_USER_TASK)).isEqualTo(5);
+    }
+
+    @Test
+    public void shouldSkipUserTasksWhenProcessInstanceIsSkipped() {
+        ch.qos.logback.classic.Logger logger = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(HistoryMigrator.class);
+        logger.setLevel(Level.DEBUG);
+        // given state in c7
+        deployer.deployCamunda7Process("userTaskProcess.bpmn");
+        var processInstance = runtimeService.startProcessInstanceByKey("userTaskProcessId");
+        var task = taskService.createTaskQuery().singleResult();
+        taskService.complete(task.getId());
+
+        // and the process instance is manually set as skipped
+        dbClient.insert(processInstance.getId(), null, IdKeyMapper.TYPE.HISTORY_PROCESS_INSTANCE);
```

Comment:
It's not ideal to break the black box test principle here by calling an internal API. However, I don't have a better idea how to test this more real-world-y.